test/runtests.jl

```diff
@@ -22,6 +22,22 @@ abstract type Iterable end
         eltype(::Iterable) || eltype(::Type{Iterable})
     end
 
+    @test x begin
+        i = 0
+        state = iterate(x)
+        eT = Base.IteratorEltype(x) == Base.HasEltype() ? eltype(x) : Any
+        while state !== nothing
+            val, st = state
+            @test val isa eT
+            i += 1
+            state = iterate(x, st)
+        end
+        if Base.IteratorSize(x) == Base.HasLength()
+            @test i == length(x)
+        elseif Base.IteratorSize(x) == Base.HasShape()
+            @test i == length(x)
+        end
+    end
```

Comment:
what advantages do we get from making this part of `@interface`?

could everything here just be a put in a function?

```diff
-   @test x begin
+   function test_interface(::Type{Iterable}, x)
        i = 0
        state = iterate(x)
        eT = Base.IteratorEltype(x) == Base.HasEltype() ? eltype(x) : Any
        while state !== nothing
            val, st = state
            @test val isa eT
            i += 1
            state = iterate(x, st)
        end
        if Base.IteratorSize(x) == Base.HasLength()
            @test i == length(x)
        elseif Base.IteratorSize(x) == Base.HasShape()
            @test i == length(x)
        end
    end
    ```